### ISSUE TITLE
[Bug] Can't bootstrap galera cluster whilst using Linkerd service mesh

### ISSUE THREAD

**Issue by boedy** [mariadb-operator/mariadb-operator#570]
**Documentation**
- [x] I acknowledge that I have read the relevant [documentation](https://github.com/mariadb-operator/mariadb-operator/tree/main/docs).

**Describe the bug**
Took a while to figure out, but when using Linkerd as service mesh the Mariadb statefulset will not be created. The only thing that runs is the Init job. After that nothing happens.

**Expected behaviour**
Cluster should bootstrap / create the statefulset

**Steps to reproduce the bug**
1. install Linkerd Service mesh
2. enable proxy injection on the namespace: `kubectl annotate ns <namespace> linkerd.io/inject=enabled`
3. Deploy galera manifest to namespace

**Debug information**
relevant operator logs
```
{"level":"debug","ts":1713731085.565182,"msg":"unable to get healthy Pod client","controller":"maxscale","controllerGroup":"k8s.mariadb.com","controllerKind":"MaxScale","MaxScale":{"name":"galera-maxscale","namespace":"analytics"},"namespace":"analytics","name":"galera-maxscale","reconcileID":"7fe8dcb3-85a9-4ae1-b877-f94cc238dcab","err":"error getting healthy Pod: no healthy instances available"}
{"level":"debug","ts":1713731085.6260598,"msg":"User not ready. Requeuing","controller":"maxscale","controllerGroup":"k8s.mariadb.com","controllerKind":"MaxScale","MaxScale":{"name":"galera-maxscale","namespace":"analytics"},"namespace":"analytics","name":"galera-maxscale","reconcileID":"7fe8dcb3-85a9-4ae1-b877-f94cc238dcab","user":"galera-maxscale-client"}
{"level":"debug","ts":1713731085.8258617,"msg":"error getting MaxScale primary Pod","controller":"mariadb","controllerGroup":"k8s.mariadb.com","controllerKind":"MariaDB","MariaDB":{"name":"galera","namespace":"analytics"},"namespace":"analytics","name":"galera","reconcileID":"c4f7c470-effe-4346-a89f-34ead85d3ed2","err":"MaxScale primary server not found"}
{"level":"debug","ts":1713731085.953008,"msg":"Init job not completed. Requeuing","controller":"mariadb","controllerGroup":"k8s.mariadb.com","controllerKind":"MariaDB","MariaDB":{"name":"galera","namespace":"analytics"},"namespace":"analytics","name":"galera","reconcileID":"c4f7c470-effe-4346-a89f-34ead85d3ed2"}
```

**Environment details**:
- Kubernetes version: 1.25
- Kubernetes distribution: k3s
- mariadb-operator version: v0.0.27
- Install method: helm

**Comment by mmontes11** [comment]
Hey there @boedy ! Thanks for reporting.

This is happening because the Galera init job hasn't completed. Linkerd injects its sidecar container by default in every `Pod`, which continues to run after the main job container has completed. 

We have addressed this situation in `v0.0.28`, but it hasn't been released yet.

As a preview, you can see how to solve this problem with `v0.0.28` with Istio, which has the same problem as LinkerD:
https://github.com/mariadb-operator/mariadb-operator/blob/6aeb483750cdc5aaeae23a18ba019089707ce9f4/examples/manifests/mariadb_galera_full.yaml#L72

Basically, avoid adding the sidecar container to the Galera init Job, to make sure it finishes and therefore the operator carries on provisioning `MariaDB` after it is done.

Closing this! Stay tuned for `v0.0.28`
